### DISCUSSION TITLE
Detect language in the first 60 seconds instead of 30 seconds

### DISCUSSION THREAD

**Discussion by toanhn** [openai/whisper#2375] [Q&A]
Hi everyone,
Whisper detects language in the first 30 seconds of an audio file. How can I force it to detect language in the first 60 seconds?
Thanks.

**Comment by jhj0517** [comment]
As far as I know whisper doesn't have such a parameter yet, but [faster-whisper](https://github.com/SYSTRAN/faster-whisper) does. 
You can use `language_detection_segments`. 

```
from faster_whisper import WhisperModel

audio_path = "test.mp3"

model = WhisperModel("tiny")
segments, info = model.transcribe(audio_path, language_detection_segments=2)
```
